src/store.rs

```diff
@@ -179,59 +185,59 @@ impl StreamWrites {
         }
     }
 
-    /// Append an event to a stream using builder pattern.
-    ///
-    /// This method consumes self and returns a new StreamWrites with the event added.
-    /// Follows immutable builder pattern for type-safe construction.
-    ///
-    /// The stream ID is extracted from the event itself via the Event trait's
-    /// stream_id() method. This ensures type safety: events know their own
-    /// aggregate identity and cannot be appended to the wrong stream.
-    ///
-    /// # Multiple Events to Same Stream
-    ///
-    /// If multiple events are appended to the same stream with different expected
-    /// versions, the last expected_version wins. This is intentional: commands that
-    /// emit multiple events to the same stream should use the same expected version
-    /// for all events (the version before any events are written).
-    ///
-    /// # Parameters
-    ///
-    /// * `event` - The event to append (must implement Event trait)
-    /// * `expected_version` - The version this write expects the stream to be at
-    ///
-    /// # Returns
+    /// Register a stream and its expected version prior to appending events.
     ///
-    /// New StreamWrites instance with the event added
-    pub fn append<E: Event>(
-        mut self,
-        event: E,
+    /// Commands must declare their optimistic concurrency expectation for each
+    /// stream before appending events. The stream identifier must match one of
+    /// the streams declared by the command's workflow.
+    pub fn register_stream(
+        self,
+        stream_id: StreamId,
         expected_version: StreamVersion,
     ) -> Result<Self, EventStoreError> {
         use std::collections::hash_map::Entry;
```

Comment:
Should this be moved to the top of the module?

Reply:
Moved `StreamWrites` (and its impl) to the top of the module in 6a41667733064652509e0a99a9fb2e3ff6f98a66 so the helper is visible right away.